File: clojure/leap/src/leap.clj
(ns leap)

(defn- divisible? [n divisor]
  (= 0 (rem n divisor)))

(defn leap-year? [y]
  (and (divisible? y 4)
       (or (divisible? y 400)
           (not (divisible? y 100)))))

File: scala/leap/src/main/scala/Leap.scala
object Leap {
  def isDivisibleBy(num: Int, divisor: Int): Boolean = {
    num % divisor == 0
  }

  def leapYear(year: Int): Boolean = {
    isDivisibleBy(year, 4) && (!isDivisibleBy(year, 100) || isDivisibleBy(year, 400))
  }
}
